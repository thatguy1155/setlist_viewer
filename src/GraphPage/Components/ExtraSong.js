import React, { useContext, useState } from 'react';
import { AppContext } from '../../Context/Context';

export default function ExtraSong(props) {
  const { artist, submit, suggest } = props;

  const [song, setSong] = useState('');
  const readyForApi = (string) => string.toLowerCase().replace(' ', '%20');

  const handleClick = () => {
    const apiReadyArtist = readyForApi(artist);
    submit({ artist: apiReadyArtist, song });
  };

  const handleChangeSong = async (event) => {
    const apiReadySong = readyForApi(event.target.value);
    setSong(apiReadySong);
    try {
      await suggest({ artist: readyForApi(artist), song: apiReadySong });
    } catch (e) {
      console.log(e);
    }
  };

  return (
    <div className>
      <label htmlFor="song">
        Song:
        <input type="text" id="song" data-testid="song" className="search-bar" onChange={handleChangeSong} />
      </label>
      <button type="button" className="submit-button" onClick={handleClick}>Submit</button>
    </div>
  );
}
