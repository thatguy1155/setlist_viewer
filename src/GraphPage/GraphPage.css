.loading{
    height:100px;
    width:100px;
}
.song-title-wrapper{
  display:flex;
  justify-content: space-between;
  padding:10px;
}

.song-title-text{
  color:white;
}