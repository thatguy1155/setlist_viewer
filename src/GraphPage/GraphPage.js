import React, { useState, useContext, useEffect } from 'react';
import {
  Container, Row, Col,
} from 'react-bootstrap';
import { AppContext } from '../Context/Context';
import Graph from './Components/Graph';
import Loading from '../Components/Loading/Loading';
import ExtraSong from './Components/ExtraSong';
import './GraphPage.css';
import { Link, useHistory } from 'react-router-dom';

export default function GraphPage() {
  const history = useHistory();
  const {
    artist, isLoading, memoizedPaddedSongs, years, error, search,
  } = useContext(AppContext);
  const emptyPage = memoizedPaddedSongs.length < 1 && !isLoading;
  const loadingPage = isLoading && memoizedPaddedSongs.length < 1;
  const loadingAdditionalSong = isLoading && memoizedPaddedSongs.length > 0;

  useEffect(() => {
    console.log(error);
    if (emptyPage && !error) history.push('/');
    // eslint-disable-line
  }, []);
  useEffect(() => {
    if (error) history.push('/error');
    // eslint-disable-line
  }, [error]);

  return (
    <Container fluid>
      <Row>
        <Col>
          {loadingPage ? <Loading /> : <Graph songs={memoizedPaddedSongs} years={years} />}
        </Col>
      </Row>
      <Row>
        <Col>
          {!loadingPage && <ExtraSong artist={artist} submit={search} />}
        </Col>
      </Row>
    </Container>
  );
}
