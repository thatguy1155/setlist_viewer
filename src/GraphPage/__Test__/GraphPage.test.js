import {
  act, render, screen, fireEvent,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import GraphPage from '../GraphPage';
import ExtraSong from '../Components/ExtraSong';
import AppContextProvider from '../../Context/Context';
import TestRenderer from 'react-test-renderer';

describe('My GraphPage', () => {
  const callback = jest.fn();
  const tally = [{ song: ['20-11-1990'] }];
  const isLoaded = true;
  test('non-shallow render', () => {
    // eslint-disable-next-line new-cap
    const element = new TestRenderer.create(
      <AppContextProvider value={{ tally, isLoaded }}>
        <GraphPage />
      </AppContextProvider>,
    );
    expect(element.toJSON()).toMatchSnapshot();
  });

  it('should return string ready for the API', () => {
    const form = render(<ExtraSong value="" search={callback} />);
    fireEvent.change(form.getByTestId('song'), {
      target: { value: 'truckin' },
    });
    act(() => {
      userEvent.click(screen.getByRole('button'));
    });

    expect(callback).toHaveBeenCalledTimes(1);
    expect(callback).toHaveBeenCalledWith({ song: 'truckin' });
  });

  //   it('should show an altered name', () => {
  //     //await screen.findByText(/Signed in as/);
  //     render(<NameDisplay name={'Slim Shady'} setName={() => {}}/>);
  //     expect(screen.queryByText(/Searches for Slim Shady/)).toBeNull();

  //     userEvent.type(screen.getByRole('textbox'), 'Slim Shady');

  //     expect(
  //       screen.getByText(/my name is Slim Shady/),
  //     ).toBeInTheDocument();
  // });
});
