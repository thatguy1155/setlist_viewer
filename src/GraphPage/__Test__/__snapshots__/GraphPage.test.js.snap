// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My GraphPage non-shallow render 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        <canvas
          data-testid="canvas"
          height={150}
          onClick={[Function]}
          role="img"
          width={300}
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className={true}
      >
        <label
          htmlFor="song"
        >
          Song:
          <input
            className="search-bar"
            data-testid="song"
            id="song"
            onChange={[Function]}
            type="text"
          />
        </label>
        <button
          className="submit-button"
          onClick={[Function]}
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
